### DISCUSSION TITLE
Is it possible to reuse an Image?

### DISCUSSION THREAD

**Discussion by jorgefm** [testcontainers/testcontainers-dotnet#1178] [Q&A] [answered]
Hi,

In my project I have the Dockerfile to build the image to use to run some integration tests. Something like:

`
public sealed class MyIntegrationTest : IAsyncLifetime
{
    private readonly IFutureDockerImage _image = null;

    private readonly string _imageName = "integration_test_image:latest";
    
    public MyIntegrationTest()
    {
        _image = new ImageFromDockerfileBuilder()
            .WithDockerfileDirectory(CommonDirectoryPath.GetSolutionDirectory(), Directory.GetCurrentDirectory())
            .WithDockerfile("Dockerfile")
            .WithName(_imageName)
            .WithCleanUp(false)
            .Build();
    }

    public async Task InitializeAsync()
    {
        await _image.CreateAsync()
            .ConfigureAwait(false);
    }

    public Task DisposeAsync()
    {
        return Task.CompletedTask;
    }

    [Fact]
    ...
`

Then every time I run a test with the given image is created and I found in my system a new integration_test_image:latest and the old image renamed to <none>:<none>

Is there some way to avoid creating the new image if there is an image already present with the same name and version?

Thanks!

**Comment by HofmeisterAn** [comment, marked as answer]
You can configure the behavior with the following API:

```csharp
WithImageBuildPolicy(Func<ImageInspectResponse, bool>)
```

By default, the builder [always](https://github.com/testcontainers/testcontainers-dotnet/blob/4d791f40361d95209e21d163687366684041414f/src/Testcontainers/Builders/ImageFromDockerfileBuilder.cs#L112C26-L112C65) builds the image. You can either use the existing pull policy predicate:

```csharp
WithImageBuildPolicy(PullPolicy.Missing)
```

or use your own:

```csharp
WithImageBuildPolicy(cachedImage => cachedImage == null)
```

I noticed that this API is not mentioned in the docs. I will update it later.

**Comment by jorgefm** [comment]
Yes, that's make the trick...

Thanks!
